Cygwin/Newlib headers don't handle feature tests properly

--- origsrc/dia-0.97.2/app/load_save.c	2011-09-25 06:55:11.000000000 -0500
+++ src/dia-0.97.2/app/load_save.c	2013-10-09 04:26:50.124841600 -0500
@@ -19,6 +19,7 @@
 
 #define _BSD_SOURCE 1 /* to get the prototype for fchmod() from sys/stat.h */
 #define _POSIX_C_SOURCE 2 /* so we get fdopen declared even when compiling with -ansi */
+#undef __STRICT_ANSI__
 #include <sys/stat.h>
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
--- origsrc/dia-0.97.2/lib/arrows.c	2011-01-07 06:01:51.000000000 -0600
+++ src/dia-0.97.2/lib/arrows.c	2013-10-09 04:03:27.951641900 -0500
@@ -19,6 +19,7 @@
 #include <config.h>
 
 #define _BSD_SOURCE 1 /* to get finite */
+#undef __STRICT_ANSI__
 #include <math.h>
 #include <stdio.h>
 #include <string.h>
--- origsrc/dia-0.97.2/lib/boundingbox.c	2009-11-07 11:13:53.000000000 -0600
+++ src/dia-0.97.2/lib/boundingbox.c	2013-10-09 04:25:30.417282600 -0500
@@ -22,6 +22,7 @@
 #include <config.h>
 
 #define _BSD_SOURCE 1
+#undef __STRICT_ANSI__
 #include <math.h>
 #include <string.h> /* memcmp() */
 
--- origsrc/dia-0.97.2/objects/custom/shape_typeinfo.c	2013-10-04 02:37:12.788166000 -0500
+++ src/dia-0.97.2/objects/custom/shape_typeinfo.c	2013-10-09 04:31:38.709347700 -0500
@@ -28,6 +28,7 @@
 #include <string.h>
 #include <stdarg.h>
 #include <glib.h>
+#include <glib/gstdio.h>
 #include <libxml/parser.h>
 
 /*
--- origsrc/dia-0.97.2/objects/standard/arc.c	2009-11-07 11:13:53.000000000 -0600
+++ src/dia-0.97.2/objects/standard/arc.c	2013-10-09 04:28:06.303198700 -0500
@@ -22,6 +22,7 @@
 
 #include <assert.h>
 #define _BSD_SOURCE 1 /* to get finite */
+#undef __STRICT_ANSI__
 #include <math.h>
 
 #include "intl.h"
--- origsrc/dia-0.97.2/plug-ins/postscript/paginate_psprint.c	2009-11-07 08:28:34.000000000 -0600
+++ src/dia-0.97.2/plug-ins/postscript/paginate_psprint.c	2013-10-09 10:21:08.608757300 -0500
@@ -25,6 +25,7 @@
 
 /* so we get popen and sigaction even when compiling with -ansi */
 #define _POSIX_C_SOURCE 2
+#undef __STRICT_ANSI__
 #include <stdio.h>
 #include <string.h> /* strlen */
 #include <signal.h>
