--- origsrc/dia-0.96.1/acinclude.m4	2007-01-06 16:39:36.000000000 -0600
+++ src/dia-0.96.1/acinclude.m4	2007-03-29 12:27:48.656250000 -0500
@@ -223,7 +223,7 @@
 [AC_REQUIRE([AM_PATH_PYTHON])
 AC_REQUIRE([AM_CHECK_PYTHON_HEADERS])
 
-AC_MSG_CHECKING(for libpython${PYTHON_VERSION}.a)
+AC_MSG_CHECKING(for libpython${PYTHON_VERSION})
 
 py_config_dir="$py_prefix/lib/python${PYTHON_VERSION}/config"
 
@@ -235,7 +235,7 @@
   py_other_libs=`sed -n -e 's/^LIBS=\(.*\)/\1/p' $py_makefile`
 
 dnl now the actual libpython
-  if test -e "${py_config_dir}/libpython${PYTHON_VERSION}.a"; then
+  if test -e "${py_config_dir}/libpython${PYTHON_VERSION}.a" -o -e "${py_config_dir}/libpython${PYTHON_VERSION}.dll.a"; then
     PYTHON_LIBS="-L${py_config_dir} -lpython${PYTHON_VERSION} $py_localmodlibs $py_basemodlibs $py_other_libs"
     AC_MSG_RESULT(found)
   else
--- origsrc/dia-0.96.1/lib/Makefile.am	2007-01-18 10:49:56.000000000 -0600
+++ src/dia-0.96.1/lib/Makefile.am	2007-03-29 12:27:48.671875000 -0500
@@ -27,7 +27,7 @@
 	propdialogs.c  \
 	propregistry.c 
 
-pkglib_LTLIBRARIES =     libdia.la
+lib_LTLIBRARIES =     libdia.la
 
 ICON_PNG_PAIRS= \
 	dia_broken_icon $(srcdir)/pixmaps/broken.png \
@@ -167,7 +167,9 @@
 #		render.h \
 #		render.c 
 
-libdia_la_LDFLAGS = -avoid-version
+libdia_la_LDFLAGS = -avoid-version -no-undefined
+
+libdia_la_LIBADD = $(GNOME_LIBS) $(GTK_LIBS)
 
 AM_CPPFLAGS = \
 	$(LIBART_CFLAGS) \
--- origsrc/dia-0.96.1/lib/plug-ins.c	2007-01-06 16:28:26.000000000 -0600
+++ src/dia-0.96.1/lib/plug-ins.c	2007-03-29 12:27:48.671875000 -0500
@@ -46,7 +46,7 @@
 #include <io.h> /* open, close */
 #endif
 
-#if defined(G_OS_WIN32) || defined(__EMX__)
+#if defined(G_PLATFORM_WIN32) || defined(__EMX__)
 #  define PLUG_IN_EXT ".dll"
 #  define PLUG_IN_EXT_LEN 4
 #  define USING_LIBTOOL 0
--- origsrc/dia-0.96.1/objects/AADL/Makefile.am	2007-01-06 16:28:08.000000000 -0600
+++ src/dia-0.96.1/objects/AADL/Makefile.am	2007-03-29 12:27:48.687500000 -0500
@@ -21,7 +21,8 @@
                              edit_port_declaration.h \
                              edit_port_declaration.c 
 
-libaadl_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libaadl_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libaadl_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/ER/Makefile.am	2007-01-06 16:28:09.000000000 -0600
+++ src/dia-0.96.1/objects/ER/Makefile.am	2007-03-29 12:27:48.687500000 -0500
@@ -8,7 +8,8 @@
 			participation.c \
 			er.c
 
-liber_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+liber_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+liber_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/FS/Makefile.am	2007-01-06 16:28:09.000000000 -0600
+++ src/dia-0.96.1/objects/FS/Makefile.am	2007-03-29 12:27:48.703125000 -0500
@@ -7,7 +7,8 @@
 			flow-ortho.c \
 			function.c
 
-libfs_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libfs_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libfs_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/GRAFCET/Makefile.am	2007-01-06 16:28:10.000000000 -0600
+++ src/dia-0.96.1/objects/GRAFCET/Makefile.am	2007-03-29 12:27:48.703125000 -0500
@@ -15,7 +15,8 @@
 			vergent.c \
 			vector.c
 
-libgrafcet_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libgrafcet_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libgrafcet_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS) $(XML_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/Istar/Makefile.am	2007-01-06 16:28:02.000000000 -0600
+++ src/dia-0.96.1/objects/Istar/Makefile.am	2007-03-29 12:27:48.953125000 -0500
@@ -9,7 +9,8 @@
 			other.c \
 			link.c
 
-libistar_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libistar_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libistar_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(FREETYPE_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/Jackson/Makefile.am	2007-01-06 16:28:02.000000000 -0600
+++ src/dia-0.96.1/objects/Jackson/Makefile.am	2007-03-29 12:27:48.968750000 -0500
@@ -8,7 +8,8 @@
 			requirement.c \
 			phenomenon.c
 
-libjackson_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libjackson_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libjackson_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(FREETYPE_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/KAOS/Makefile.am	2007-01-06 16:28:06.000000000 -0600
+++ src/dia-0.96.1/objects/KAOS/Makefile.am	2007-03-29 12:27:48.968750000 -0500
@@ -9,7 +9,8 @@
 			metabinrel.c \
 			metaandorrel.c
 
-libkaos_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libkaos_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libkaos_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(FREETYPE_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/Misc/Makefile.am	2007-01-06 16:28:03.000000000 -0600
+++ src/dia-0.96.1/objects/Misc/Makefile.am	2007-03-29 12:27:48.968750000 -0500
@@ -6,7 +6,8 @@
 			analog_clock.c
 
 
-libmisc_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libmisc_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libmisc_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/Misc/libmisc.c	2007-01-06 16:28:03.000000000 -0600
+++ src/dia-0.96.1/objects/Misc/libmisc.c	2007-03-29 12:27:49.015625000 -0500
@@ -29,7 +29,7 @@
 #include "plug-ins.h"
 
 extern DiaObjectType analog_clock_type;
-#ifndef _MSC_VER
+#ifndef G_PLATFORM_WIN32
 /** Taken from app, but here while testing. */
 extern DiaObjectType newgroup_type;
 #endif
@@ -44,7 +44,7 @@
     return DIA_PLUGIN_INIT_ERROR;
 
   object_register_type(&analog_clock_type);  
-#ifndef _MSC_VER
+#ifndef G_PLATFORM_WIN32
   object_register_type(&newgroup_type);
 #endif
 
--- origsrc/dia-0.96.1/objects/SADT/Makefile.am	2007-01-06 16:28:03.000000000 -0600
+++ src/dia-0.96.1/objects/SADT/Makefile.am	2007-03-29 12:27:49.015625000 -0500
@@ -8,7 +8,8 @@
 			box.c \
 			annotation.c 
 
-libsadt_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libsadt_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libsadt_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/UML/Makefile.am	2007-01-06 16:28:06.000000000 -0600
+++ src/dia-0.96.1/objects/UML/Makefile.am	2007-03-29 12:27:49.031250000 -0500
@@ -38,7 +38,8 @@
 			umlparameter.c \
 			umlformalparameter.c
 
-libuml_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libuml_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libuml_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/bondgraph/Makefile.am	2007-01-06 16:28:08.000000000 -0600
+++ src/dia-0.96.1/objects/bondgraph/Makefile.am	2007-03-29 12:27:49.031250000 -0500
@@ -4,7 +4,8 @@
 libbondgraph_objects_la_SOURCES = \
 			bondgraph.c
 
-libbondgraph_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libbondgraph_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libbondgraph_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) \
 	$(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
--- origsrc/dia-0.96.1/objects/chronogram/Makefile.am	2007-01-06 16:28:06.000000000 -0600
+++ src/dia-0.96.1/objects/chronogram/Makefile.am	2007-03-29 12:27:49.046875000 -0500
@@ -10,7 +10,8 @@
 			chronoline_event.c
 
 
-libchronogram_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libchronogram_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libchronogram_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/custom/Makefile.am	2007-01-06 16:28:09.000000000 -0600
+++ src/dia-0.96.1/objects/custom/Makefile.am	2007-03-29 12:27:49.062500000 -0500
@@ -10,7 +10,8 @@
 	custom_util.h \
 	custom.c
 
-libcustom_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libcustom_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libcustom_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 shapedir = $(pkgdatadir)/shapes
 
--- origsrc/dia-0.96.1/objects/flowchart/Makefile.am	2007-01-06 16:28:02.000000000 -0600
+++ src/dia-0.96.1/objects/flowchart/Makefile.am	2007-03-29 12:27:49.062500000 -0500
@@ -8,7 +8,8 @@
 			ellipse.c \
 			flowchart.c
 
-libflowchart_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libflowchart_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libflowchart_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) \
 	$(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
--- origsrc/dia-0.96.1/objects/network/Makefile.am	2007-01-06 16:28:03.000000000 -0600
+++ src/dia-0.96.1/objects/network/Makefile.am	2007-03-29 12:27:49.078125000 -0500
@@ -9,7 +9,8 @@
 			      basestation.c \
 			      wanlink.c 
 
-libnetwork_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libnetwork_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libnetwork_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
 
--- origsrc/dia-0.96.1/objects/standard/Makefile.am	2007-01-06 16:28:08.000000000 -0600
+++ src/dia-0.96.1/objects/standard/Makefile.am	2007-03-29 12:27:49.078125000 -0500
@@ -14,7 +14,8 @@
 			beziergon.c \
 			tool-icons.h
 
-libstandard_objects_la_LDFLAGS = -export-dynamic -module -avoid-version
+libstandard_objects_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libstandard_objects_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 ICON_PNG_PAIRS= \
 	arc_icon $(srcdir)/pixmaps/arc.png \
--- origsrc/dia-0.96.1/plug-ins/cairo/Makefile.am	2007-01-06 16:39:29.000000000 -0600
+++ src/dia-0.96.1/plug-ins/cairo/Makefile.am	2007-03-29 12:27:49.125000000 -0500
@@ -7,8 +7,8 @@
 
 libcairo_filter_la_SOURCES = $(plugin_sources)
 
-libcairo_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libcairo_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
 
-libcairo_filter_la_LIBADD = $(CAIRO_LIBS)
+libcairo_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS) $(CAIRO_LIBS)
 
 INCLUDES = -I$(srcdir)/../../lib $(GTK_CFLAGS) $(DEBUG_FLAGS) $(CAIRO_CFLAGS)
--- origsrc/dia-0.96.1/plug-ins/cgm/Makefile.am	2007-01-06 16:39:32.000000000 -0600
+++ src/dia-0.96.1/plug-ins/cgm/Makefile.am	2007-03-29 12:27:49.125000000 -0500
@@ -4,7 +4,8 @@
 
 libcgm_filter_la_SOURCES = cgm.c
 
-libcgm_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libcgm_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libcgm_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) \
 	$(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
--- origsrc/dia-0.96.1/plug-ins/dxf/Makefile.am	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/dxf/Makefile.am	2007-03-29 12:27:49.125000000 -0500
@@ -8,7 +8,8 @@
 	dxf-import.c \
 	autocad_pal.h
 
-libdxf_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libdxf_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libdxf_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) \
 	$(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(XML_CFLAGS) $(UNICODE_CFLAGS) 
--- origsrc/dia-0.96.1/plug-ins/gprint/Makefile.am	2007-01-06 16:39:31.000000000 -0600
+++ src/dia-0.96.1/plug-ins/gprint/Makefile.am	2007-03-29 12:27:49.140625000 -0500
@@ -12,8 +12,8 @@
 
 libgprint_filter_la_SOURCES = $(plugin_sources)
 
-libgprint_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libgprint_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
 
-libgprint_filter_la_LIBADD = $(GTK_LIBS) $(GNOMEPRINT_LIBS)
+libgprint_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS) $(GNOMEPRINT_LIBS)
 
 INCLUDES = -I$(srcdir)/../../lib $(GTK_CFLAGS) $(DEBUG_FLAGS) $(GNOMEPRINT_CFLAGS)
--- origsrc/dia-0.96.1/plug-ins/hpgl/Makefile.am	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/hpgl/Makefile.am	2007-03-29 12:27:49.171875000 -0500
@@ -1,7 +1,8 @@
 pkglib_LTLIBRARIES = libhpgl_filter.la
 
 libhpgl_filter_la_SOURCES = hpgl.c
-libhpgl_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libhpgl_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libhpgl_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(top_srcdir)/lib \
   $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS)
--- origsrc/dia-0.96.1/plug-ins/metapost/Makefile.am	2007-01-06 16:39:29.000000000 -0600
+++ src/dia-0.96.1/plug-ins/metapost/Makefile.am	2007-03-29 12:27:49.187500000 -0500
@@ -7,7 +7,8 @@
 	render_metapost.c \
 	render_metapost.h
 
-libmetapost_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libmetapost_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libmetapost_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib \
  $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(DEBUG_CFLAGS) \
--- origsrc/dia-0.96.1/plug-ins/pgf/Makefile.am	2007-01-06 16:39:32.000000000 -0600
+++ src/dia-0.96.1/plug-ins/pgf/Makefile.am	2007-03-29 12:27:49.187500000 -0500
@@ -7,7 +7,8 @@
 	render_pgf.c \
 	render_pgf.h
 
-libpgf_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libpgf_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libpgf_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) \
 	$(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS) $(DEBUG_CFLAGS)
--- origsrc/dia-0.96.1/plug-ins/pixbuf/Makefile.am	2007-01-06 16:39:29.000000000 -0600
+++ src/dia-0.96.1/plug-ins/pixbuf/Makefile.am	2007-03-29 12:27:49.203125000 -0500
@@ -5,6 +5,7 @@
 libpixbuf_filter_la_SOURCES = \
 	pixbuf.c
 
-libpixbuf_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libpixbuf_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libpixbuf_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(DEBUG_FLAGS)
--- origsrc/dia-0.96.1/plug-ins/pstricks/Makefile.am	2007-01-06 16:39:31.000000000 -0600
+++ src/dia-0.96.1/plug-ins/pstricks/Makefile.am	2007-03-29 12:27:49.250000000 -0500
@@ -7,7 +7,8 @@
 	render_pstricks.c \
 	render_pstricks.h
 
-libpstricks_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libpstricks_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libpstricks_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) \
 	$(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS) $(DEBUG_CFLAGS)
--- origsrc/dia-0.96.1/plug-ins/python/Makefile.am	2007-03-17 15:19:12.000000000 -0500
+++ src/dia-0.96.1/plug-ins/python/Makefile.am	2007-03-29 12:27:49.250000000 -0500
@@ -14,8 +14,8 @@
 
 pkglib_LTLIBRARIES = $(libpython_plugin)
 
-libpython_plugin_la_LDFLAGS = -export-dynamic -module -avoid-version
-libpython_plugin_la_LIBADD = $(PYTHON_LIBS)
+libpython_plugin_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libpython_plugin_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS) $(PYTHON_LIBS)
 libpython_plugin_la_SOURCES = \
 	pydia-color.c \
 	pydia-color.h \
--- origsrc/dia-0.96.1/plug-ins/python/pydia-color.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-color.c	2007-03-29 12:27:49.265625000 -0500
@@ -104,7 +104,7 @@
  * Python objetcs
  */
 PyTypeObject PyDiaColor_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "DiaColor",
     sizeof(PyDiaColor),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-cpoint.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-cpoint.c	2007-03-29 12:27:49.265625000 -0500
@@ -81,7 +81,7 @@
 }
 
 PyTypeObject PyDiaConnectionPoint_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.ConnectionPoint",
     sizeof(PyDiaConnectionPoint),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-diagram.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-diagram.c	2007-03-29 12:27:49.281250000 -0500
@@ -527,7 +527,7 @@
 }
 
 PyTypeObject PyDiaDiagram_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Diagram",
     sizeof(PyDiaDiagram),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-diagramdata.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-diagramdata.c	2007-03-29 12:27:49.281250000 -0500
@@ -377,7 +377,7 @@
 }
 
 PyTypeObject PyDiaDiagramData_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.DiagramData",
     sizeof(PyDiaDiagramData),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-display.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-display.c	2007-03-29 12:27:49.296875000 -0500
@@ -233,7 +233,7 @@
 }
 
 PyTypeObject PyDiaDisplay_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Display",
     sizeof(PyDiaDisplay),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-error.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-error.c	2007-03-29 12:27:49.296875000 -0500
@@ -160,7 +160,7 @@
  * Python objetcs
  */
 PyTypeObject PyDiaError_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "DiaError",
     sizeof(PyDiaError),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-export.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-export.c	2007-03-29 12:27:49.312500000 -0500
@@ -80,7 +80,7 @@
 }
 
 PyTypeObject PyDiaExportFilter_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.ExportFilter",
     sizeof(PyDiaExportFilter),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-font.c	2007-03-17 15:19:12.000000000 -0500
+++ src/dia-0.96.1/plug-ins/python/pydia-font.c	2007-03-29 12:27:49.312500000 -0500
@@ -126,7 +126,7 @@
  */
 
 PyTypeObject PyDiaFont_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Font",
     sizeof(PyDiaFont),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-geometry.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-geometry.c	2007-03-29 12:27:49.312500000 -0500
@@ -452,7 +452,7 @@
  * Python objetcs
  */
 PyTypeObject PyDiaPoint_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Point",
     sizeof(PyDiaPoint),
@@ -477,7 +477,7 @@
 };
 
 PyTypeObject PyDiaRectangle_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Rectangle",
     sizeof(PyDiaRectangle),
@@ -502,7 +502,7 @@
 };
 
 PyTypeObject PyDiaBezPoint_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.BezPoint",
     sizeof(PyDiaBezPoint),
@@ -527,7 +527,7 @@
 };
 
 PyTypeObject PyDiaArrow_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Arrow",
     sizeof(PyDiaArrow),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-handle.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-handle.c	2007-03-29 12:27:49.328125000 -0500
@@ -112,7 +112,7 @@
 }
 
 PyTypeObject PyDiaHandle_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Handle",
     sizeof(PyDiaHandle),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-image.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-image.c	2007-03-29 12:27:49.328125000 -0500
@@ -138,7 +138,7 @@
  * Python objetcs
  */
 PyTypeObject PyDiaImage_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Image",
     sizeof(PyDiaImage),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-layer.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-layer.c	2007-03-29 12:27:49.343750000 -0500
@@ -236,7 +236,7 @@
 }
 
 PyTypeObject PyDiaLayer_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Layer",
     sizeof(PyDiaLayer),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-object.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-object.c	2007-03-29 12:27:49.359375000 -0500
@@ -211,7 +211,7 @@
 }
 
 PyTypeObject PyDiaObject_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Object",
     sizeof(PyDiaObject),
@@ -343,7 +343,7 @@
 }
 
 PyTypeObject PyDiaObjectType_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.ObjectType",
     sizeof(PyDiaObjectType),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-properties.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-properties.c	2007-03-29 12:27:49.359375000 -0500
@@ -273,7 +273,7 @@
  * Python object
  */
 PyTypeObject PyDiaProperties_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Properties",
     sizeof(PyDiaProperties),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-property.c	2007-02-21 07:36:35.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-property.c	2007-03-29 12:27:49.375000000 -0500
@@ -764,7 +764,7 @@
  * Python object
  */
 PyTypeObject PyDiaProperty_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Property",
     sizeof(PyDiaProperty),
--- origsrc/dia-0.96.1/plug-ins/python/pydia-text.c	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/python/pydia-text.c	2007-03-29 12:27:49.375000000 -0500
@@ -120,7 +120,7 @@
  * Python objetcs
  */
 PyTypeObject PyDiaText_Type = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,
     "dia.Text",
     sizeof(PyDiaText),
--- origsrc/dia-0.96.1/plug-ins/shape/Makefile.am	2007-01-06 16:39:30.000000000 -0600
+++ src/dia-0.96.1/plug-ins/shape/Makefile.am	2007-03-29 12:27:49.375000000 -0500
@@ -6,7 +6,8 @@
 	shape.c \
 	shape-export.c
 
-libshape_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libshape_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libshape_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) \
 	$(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS) $(XML_CFLAGS) \
--- origsrc/dia-0.96.1/plug-ins/svg/Makefile.am	2007-01-06 16:39:30.000000000 -0600
+++ src/dia-0.96.1/plug-ins/svg/Makefile.am	2007-03-29 12:27:49.390625000 -0500
@@ -4,6 +4,7 @@
 
 libsvg_filter_la_SOURCES = svg.c render_svg.c svg-import.c
 
-libsvg_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libsvg_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libsvg_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS) $(XML_CFLAGS) $(DEBUG_CFLAGS)
--- origsrc/dia-0.96.1/plug-ins/vdx/Makefile.am	2007-03-19 02:40:56.000000000 -0500
+++ src/dia-0.96.1/plug-ins/vdx/Makefile.am	2007-03-29 12:27:49.421875000 -0500
@@ -11,6 +11,7 @@
 	vdx-xml.c \
 	vdx.h
 
-libvdx_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libvdx_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libvdx_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(XML_CFLAGS) $(UNICODE_CFLAGS) $(DEBUG_FLAGS)
--- origsrc/dia-0.96.1/plug-ins/wmf/Makefile.am	2007-01-06 16:39:28.000000000 -0600
+++ src/dia-0.96.1/plug-ins/wmf/Makefile.am	2007-03-29 12:27:49.437500000 -0500
@@ -2,8 +2,8 @@
 pkglib_LTLIBRARIES = libwmf_filter.la
 
 libwmf_filter_la_SOURCES = $(plugin_sources)
-libwmf_filter_la_LDFLAGS = -export-dynamic -module -avoid-version 
-libwmf_filter_la_LIBADD = -lstdc++
+libwmf_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libwmf_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS) -lstdc++ -lgdi32
 
 INCLUDES = -I$(top_srcdir)/intl -I$(top_srcdir)/lib \
   $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS) \
--- origsrc/dia-0.96.1/plug-ins/wmf/wmf.cpp	2007-03-18 14:49:15.000000000 -0500
+++ src/dia-0.96.1/plug-ins/wmf/wmf.cpp	2007-03-29 12:27:49.437500000 -0500
@@ -42,7 +42,7 @@
 }
 #endif
 
-#if defined HAVE_WINDOWS_H || defined G_OS_WIN32
+#if defined G_PLATFORM_WIN32
 namespace W32 {
 // at least Rectangle conflicts ...
 #define WIN32_LEAN_AND_MEAN
@@ -53,6 +53,11 @@
 #define SAVE_EMF
 #endif
 
+/* PS_JOIN_MASK is missing from the w32api headers */
+#ifndef PS_JOIN_MASK
+#define PS_JOIN_MASK (PS_JOIN_BEVEL|PS_JOIN_MITER|PS_JOIN_ROUND)
+#endif
+
 /* force linking with gdi32 */
 #pragma comment( lib, "gdi32" )
 
@@ -139,7 +144,7 @@
     W32::HPEN hOldPen;
     if (colour) {
 	W32::COLORREF rgb = W32COLOR(colour);
-#ifdef G_OS_WIN32
+#ifdef G_PLATFORM_WIN32
 	if ((renderer->platform_is_nt && renderer->hPrintDC) || renderer->target_emf) {
           W32::LOGBRUSH logbrush;
 	  W32::DWORD    dashes[6];
@@ -184,7 +189,7 @@
 	    &logbrush, num_dashes, num_dashes > 0 ? dashes : 0);
 	}
 	else 
-#endif /* G_OS_WIN32 */
+#endif /* G_PLATFORM_WIN32 */
 	{
 	  renderer->hPen = W32::CreatePen(renderer->fnPenStyle, 
 					  renderer->nLineWidth,
@@ -261,7 +266,7 @@
     DIAG_NOTE(renderer, "end_render\n");
     hEmf = W32::CloseEnhMetaFile(renderer->hFileDC);
 
-#if !defined SAVE_EMF && defined G_OS_WIN32 /* the later offers both */
+#if !defined SAVE_EMF && defined G_PLATFORM_WIN32 /* the later offers both */
     /* Don't do it when printing */
     if (renderer->sFileName && strlen (renderer->sFileName)) {
 
@@ -893,7 +898,7 @@
             gunichar2* swc = g_utf8_to_utf16 (text, -1, NULL, &wclen, NULL);
             W32::TextOutW (renderer->hFileDC,
                            SCX(pos->x), SCY(pos->y),
-                           swc, wclen);
+                           (W32::WCHAR*)swc, wclen);
             g_free (swc);
         }
     }
@@ -1298,7 +1303,7 @@
                               NULL, NULL))
         return DIA_PLUGIN_INIT_ERROR;
 
-#ifdef G_OS_WIN32
+#ifdef G_PLATFORM_WIN32
     /*
      * On non windoze platforms this plug-in currently is only
      * useful at compile/develoment time. The output is broken
--- origsrc/dia-0.96.1/plug-ins/wpg/Makefile.am	2007-01-06 16:39:29.000000000 -0600
+++ src/dia-0.96.1/plug-ins/wpg/Makefile.am	2007-03-29 12:27:49.453125000 -0500
@@ -1,7 +1,8 @@
 pkglib_LTLIBRARIES = libwpg_filter.la
 
 libwpg_filter_la_SOURCES = wpg.c wpg_defs.h
-libwpg_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libwpg_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libwpg_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(top_srcdir)/lib \
   $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(UNICODE_CFLAGS) \
--- origsrc/dia-0.96.1/plug-ins/xfig/Makefile.am	2007-01-06 16:39:30.000000000 -0600
+++ src/dia-0.96.1/plug-ins/xfig/Makefile.am	2007-03-29 12:27:49.453125000 -0500
@@ -9,6 +9,7 @@
 	xfig-common.c \
 	xfig.h
 
-libxfig_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libxfig_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
+libxfig_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(XML_CFLAGS) $(UNICODE_CFLAGS) $(DEBUG_FLAGS)
--- origsrc/dia-0.96.1/plug-ins/xslt/Makefile.am	2007-01-06 16:39:29.000000000 -0600
+++ src/dia-0.96.1/plug-ins/xslt/Makefile.am	2007-03-29 12:27:49.484375000 -0500
@@ -21,8 +21,8 @@
 
 libxslt_filter_la_SOURCES = $(plugin_sources)
 
-libxslt_filter_la_LDFLAGS = -export-dynamic -module -avoid-version
+libxslt_filter_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
 
-libxslt_filter_la_LIBADD = $(XSLT_LIBS)
+libxslt_filter_la_LIBADD = $(top_builddir)/lib/libdia.la $(GTK_LIBS) $(XSLT_LIBS)
 
 INCLUDES = -I$(top_srcdir)/intl -I$(srcdir)/../../lib $(GTK_CFLAGS) $(GNOME_CFLAGS) $(PANGOFT2_CFLAGS) $(XML_CFLAGS) $(UNICODE_CFLAGS) $(DEBUG_FLAGS) $(XSLT_CFLAGS)
